HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -2341,6 +2346,301 @@ def calc_bounding_values(self):
+                    p_variance = self.PermShkStd[i]**2 # set variance of permanent income shocks this period
+                    max_p = max_p_fac*(p_variance/(1-self.LivPrb[i]))**0.5 # Consider probability of staying alive this period
+                    one_sided_grid = make_grid_exp_mult(1.05+1e-3, np.exp(max_p), num_points, 2) 
+                    
+                    dist_pGrid = np.append(np.append(1.0/np.fliplr([one_sided_grid])[0],np.ones(1)),one_sided_grid) # Compute permanent income grid this period. Grid of permanent income may differ dependent on PermShkStd
+                    self.dist_pGrid.append(dist_pGrid)
+
+            else:
+                self.dist_pGrid = dist_pGrid #If grid of permanent income prespecified then use as pgrid
+                           
+            if self.neutral_measure == True: # Use Harmenberg 2021's Neutral Measure
+                
+                self.dist_pGrid = self.T_cycle*[np.array([1])]
+            
+                
+                
+    def calc_transition_matrix(self, shk_dstn = None):
+        '''
+        Calculates how the distribution of agents across market resources 
+        transitions from one period to the next. If finite horizon problem, then calculates
+        a list of transition matrices, consumption and asset policy grids for each period of the problem. 
+        The transition matrix/matrices and consumption and asset policy grid(s) are stored as attributes of self.
+        
+        
+        Parameters
+        ----------
+            shk_dstn: list 
+                list of income shock distributions
+        Returns
+        -------
+        None
+        
+        ''' 
+        
+        
+        if self.cycles == 0: 
+            
+            if shk_dstn == None:
+                shk_dstn = self.IncShkDstn
+            
+            dist_mGrid = self.dist_mGrid #Grid of market resources
+            dist_pGrid = self.dist_pGrid #Grid of permanent incomes
+            aNext = dist_mGrid - self.solution[0].cFunc(dist_mGrid)  #assets next period
+            
+            self.aPol_Grid = aNext # Steady State Asset Policy Grid
+            self.cPol_Grid = self.solution[0].cFunc(dist_mGrid) #Steady State Consumption Policy Grid
+            
+            # Obtain shock values and shock probabilities from income distribution
+            bNext = self.Rfree*aNext # Bank Balances next period (Interest rate * assets)
+            shk_prbs = shk_dstn[0].pmf  # Probability of shocks 
+            tran_shks = shk_dstn[0].X[1] # Transitory shocks
+            perm_shks = shk_dstn[0].X[0] # Permanent shocks
+            LivPrb = self.LivPrb[0] # Update probability of staying alive
+            
+            #New borns have this distribution (assumes start with no assets and permanent income=1)
+            NewBornDist = jump_to_grid(tran_shks,np.ones_like(tran_shks),shk_prbs,dist_mGrid,dist_pGrid)
```

Comment:
Is this a probability distribution? Would it be useful to use our `DiscreteDistribution` constructor here?

Reply:
This is a function that takes in a grid, a set of values that may lie between the gridpoints, and finally the associated probabilities of each the values and returns the probability mass of each gridpoint. I don't think itd be useful to use DiscreteDistribution here